breacharbiter.go

```diff
@@ -347,46 +390,39 @@ out:
 			b.wg.Add(1)
 			go b.exactRetribution(cfChan, breachInfo)
 
-			delete(b.breachObservers, breachInfo.chanPoint)
+			delete(breachObservers, breachInfo.chanPoint)
 
 		case chanPoint := <-b.newContracts:
 			// A new channel has just been opened within the
 			// daemon, so we launch a new breachObserver to handle
 			// the detection of attempted contract breaches.
-			settleSignal := make(chan struct{})
+			breachSignal := make(chan *ContractBreachEvent, 1)
 
 			// If the contract is already being watched, then an
 			// additional send indicates we have a stale version of
 			// the contract. So we'll cancel active watcher
 			// goroutine to create a new instance with the latest
 			// contract reference.
-			if oldSignal, ok := b.breachObservers[chanPoint]; ok {
+			if oldSignal, ok := breachObservers[chanPoint]; ok {
 				brarLog.Infof("ChannelPoint(%v) is now live, "+
 					"abandoning state contract for live "+
 					"version", chanPoint)
 				close(oldSignal)
 			}
 
-			b.breachObservers[chanPoint] = settleSignal
+			breachObservers[chanPoint] = breachSignal
```

Comment:
Similar to above, I think this whole section just needs to set the `chanPoint` in `activeChannels`? Shouldn't matter if it already exists